### PR TITLE
Add pipeline job for resource group on centos7

### PR DESCRIPTION
**Pull request by xiong-gang**
Signed-off-by: Zhenghua Lyu <zlv@pivotal.io>

### CODE REVIEW

**Comment by cjcjameson** [comment]
Switched reviewers to the toolsmiths -- neither Jingyi nor I are on toolsmiths anymore

File: concourse/scripts/ic_gpdb_resgroup.bash [hunk 1]
```diff
@@ -40,15 +59,20 @@ run_resgroup_test() {
         export MASTER_DATA_DIRECTORY=/data/gpdata/master/gpseg-1; \
 
         cd /home/gpadmin/gpdb_src; \
-        ./configure --prefix=/usr/local/greenplum-db-devel --without-zlib --without-rt --without-libcurl --without-libedit-preferred --without-docdir --without-readline --disable-gpcloud --disable-gpfdist --disable-orca ${CONFIGURE_FLAGS}; \
+        ./configure --prefix=/usr/local/greenplum-db-devel --without-zlib --without-rt --without-libcurl --without-libedit-preferred --without-docdir --without-readline --disable-gpcloud --disable-gpfdist --disable-orca --disable-pxf ${CONFIGURE_FLAGS}; \
```
Comment (cjcjameson) [inline concourse/scripts/ic_gpdb_resgroup.bash]: This snuck in -- is it a merge conflict resolution accident? Seems co-incidental to this pr

File: concourse/pipelines/pipeline.yml [hunk 2]
```diff
@@ -457,14 +459,23 @@ resources:
...
   action: create
   delete_on_failure: true
   generate_random_name: true
   terraform_source: ccp_src/aws/
+
+ccp_create_params_anchor: &ccp_create_params_centos6
+  <<: *ccp_common
+  vars:
+    aws_instance-node-instance_type: t2.medium
+    platform: centos6
```
Comment (cjcjameson) [inline concourse/pipelines/pipeline.yml]: can you point to the documentation about this `platform:` field?

As a stylistic note, the decomposition into reusable anchors is making it hard for me to follow what this refers to. Certainly, I'm not embedded in the CCP-i-fy-ing process as much as others are. But @divyabhargov can you speak to the trade-off: DRY decomposition versus readability?

**Comment by cjcjameson** [comment]
I only took a brief look at the yaml -- others should review.

What's the rationale for testing on both centos6 and centos7? Do you have specific expectations about what would be different... if so can you encode that in specific tests? For most other cases, we're only doing a limited set of pipeline pieces on centos7

**Comment by kmacoskey** [comment]
We have the same concerns right now about which pattern might work best moving forward. Reusability with anchors helps keeps the pipelines smaller, but if we go down a path of adding an anchor per CCP configuration knob then it will quickly become less maintainable.

The alternative to the pattern in the PR would be do only have one common anchor that can have defaults in it. Then override the values of the common anchor in jobs only when needed. Like so:

```
ccp_create_params_anchor: &ccp_create_common
  action: create
  delete_on_failure: true
  generate_random_name: true
  terraform_source: ccp_src/aws/
  vars:
    aws_instance-node-instance_type: t2.medium
    # This would be the default, it's debatable if a default should be specified or whether
    #   it should always be explicit
    platform: centos6
```
...

```
  - put: terraform
    params:
      <<: *ccp_create_common
     # Specifying the platform after using the anchor will override the default from the anchor
      platform: centos7
```
...

```
  - put: terraform
    params:
      <<: *ccp_create_common
     # There is some gain in being explicit
      platform: centos6
```

The platform field comes from here FYI: https://github.com/Pivotal-DataFabric/concourse-cluster-provisioner#vm-operating-system

File: concourse/scripts/ic_gpdb_resgroup.bash (hunk 1 above)
Comment (xiong-gang) [reply concourse/scripts/ic_gpdb_resgroup.bash]: yes, i think we don't need this anymore as we have change it to `--enable-pxf`

**Comment by xiong-gang** [comment]
@cjcjameson 
This job and the other job for `centos6` only run resource group tests, we expect those cases to pass on all platforms we support: `centos6`, `centos7` and `suse11`.

**Comment by xiong-gang** [comment]
@kmacoskey 
I read this answer on stackoverflow: https://stackoverflow.com/questions/4875185/override-yaml-subkey
Seems like the alternative of overriding parameters in the anchor is not supported in yaml.

**Comment by cjcjameson** [comment]
@xiong-gang yep, maybe I don't quite understand the test. I agree that I expect those tests to pass, but there are many things we expect to "just work" on centos 7 after testing on centos 6. What is it about the resource groups and the differences between OS versions that make it interesting to test? Are there particular tests that demonstrate those differences, which you could point to?

**Comment by kmacoskey** [comment]
@xiong-gang 

It might not be supported for some yaml parsers, but it does work with the one concourse uses.
My original example was slightly incorrect because you can't override a value in an anchor within a block of that block. So instead it would look like this:

```
ccp_create_params_anchor: &ccp_default_params
  action: create
  delete_on_failure: true
  generate_random_name: true
  terraform_source: ccp_src/aws/

ccp_vars_anchor: &ccp_default_vars
  aws_instance-node-instance_type: t2.medium
  platform: centos6
```
...
```
  - put: terraform
    params:
      <<: *ccp_default_params
      vars:
        <<: *ccp_default_vars
        platform: centos7
```

I think the toolsmiths are in an agreement of preferring this pattern because then there are only two anchors, the two for default settings, and not an anchor per configurable field. As we add more features to CCP, we want to avoid a proliferation of yaml anchors.

I stood up a test pipeline to verify it works as expected. 
https://gpdb.ci.pivotalci.info/teams/gpdb/pipelines/test-ccp

**Comment by xiong-gang** [comment]
@kmacoskey 
I see your point, thanks. I will change it.

**Comment by xiong-gang** [comment]
@cjcjameson 
Mostly, we expect cgroups on both platforms are ok.
So you are right, we can have centos7 run a subset of resource group tests. But as the job is relatively small (about 24 mins), i think we can run the whole test for now, at least during the developing phase.

**Pull request merged**